docs/plugin-proposal-decorators.md

```diff
@@ -156,30 +156,59 @@ Use the legacy (stage 1) decorators syntax and behavior.
 
 #### NOTE: Compatibility with `@babel/plugin-transform-class-properties`
 
-If you are including your plugins manually and using `@babel/plugin-transform-class-properties` or `@babel/plugin-private-methods` and legacy decorators, make sure that `@babel/plugin-proposal-decorators` comes _before_ `@babel/plugin-transform-class-properties`.
+If you are including your plugins manually and using class elements transforms such as
+- `@babel/plugin-transform-class-properties`
+- `@babel/plugin-transform-private-methods`
+- `@babel/plugin-transform-private-property-in-object`
+- `@babel/plugin-transform-class-static-block`
+
+make sure that `@babel/plugin-proposal-decorators` comes _before_ them.
 
 ```diff title="babel.config.json"
 {
   "plugins": [
 -   "@babel/plugin-transform-class-properties",
-    ["@babel/plugin-proposal-decorators", { "version": "legacy" }]
+    ["@babel/plugin-proposal-decorators", { "version": "2023-05" }]
 +   "@babel/plugin-transform-class-properties"
   ]
 }
 ```
 
-If you are already using `@babel/preset-env`, you can safely remove `@babel/plugin-transform-class-properties` and `@babel/plugin-private-methods`:
+If you are already using `@babel/preset-env` and Stage 3 decorators, you can safely remove the
+class elements transform, Babel will automatically apply decorators transform before any presets:
+
+```diff title="babel.config.json"
+{
+  "presets": [
+    ["@babel/preset-env"],
+  ],
+  "plugins": [
+-   "@babel/plugin-transform-class-properties",
+    ["@babel/plugin-proposal-decorators", { "version": "2023-05" }]
+  ]
+}
+```
+
+If you are using `@babel/preset-env` and legacy decorators, you must manually enable the class elements transform:
```

Comment:
```suggestion
If you are using `@babel/preset-env` and legacy decorators, you must ensure the class elements transform is enabled regardless of your targets, because Babel only supports compiling legacy decorators when also compiling class properties:
```